saltproc/depcode.py

```diff
@@ -159,20 +166,346 @@ def write_mat_file(self, dep_dict, dep_end_time):
+
+    def write_mat_file(self, dep_dict, dep_end_time):
+        """Writes the iteration input file containing the burnable materials
+        composition used in OpenMC depletion runs and updated after each
+        depletion step.
+
+        Parameters
+        ----------
+        dep_dict : dict of str to Materialflow
+            Dictionary that contains `Materialflow` objects.
+
+            ``key``
+                Name of burnable material.
+            ``value``
+                `Materialflow` object holding composition and properties.
+        dep_end_time : float
+            Current time at the end of the depletion step (d).
+
+        """
+
+    def write_saltproc_openmc_tallies(self, materials, geometry):
+        """
+        Write tallies for calculating burnup and delayed neutron
+        parameters.
+
+        Parameters
+        ----------
+        materials : `openmc.Materials` object
+            The materials for the depletion simulation
+        geometry : `openmc.Geometry` object
+            The geometry for the depletion simulation
+
+        """
+        tallies = openmc.Tallies()
+
+        tally = openmc.Tally(name='delayed-fission-neutrons')
+        tally.filters = [openmc.DelayedGroupFilter([1, 2, 3, 4, 5, 6])]
+        tally.scores = ['delayed-nu-fission']
+        tallies.append(tally)
+
+        tally = openmc.Tally(name='total-fission-neutrons')
+        tally.filters = [openmc.UniverseFilter(geometry.root_universe)]
+        tally.scores = ['nu-fission']
+        tallies.append(tally)
+
+        tally = openmc.Tally(name='precursor-decay-constants')
+        tally.filters = [openmc.DelayedGroupFilter([1, 2, 3, 4, 5, 6])]
+        tally.scores = ['decay-rate']
+        tallies.append(tally)
+
+        tally = openmc.Tally(name='fission-energy')
+        tally.filters = [openmc.UniverseFilter(geometry.root_universe)]
+        tally.scores = [
+            'fission-q-recoverable',
+            'fission-q-prompt',
+            'kappa-fission']
+        tallies.append(tally)
+
+        tally = openmc.Tally(name='normalization-factor')
+        tally.filters = [openmc.UniverseFilter(geometry.root_universe)]
+        tally.scores = ['heating']
+        tallies.append(tally)
+
+        out_path = os.path.dirname(self.iter_inputfile['settings'])
+        self.iter_inputfile['tallies'] = \
+            os.path.join(out_path, 'tallies.xml')
+        tallies.export_to_xml(self.iter_inputfile['tallies'])
+        del tallies
+
+
 class DepcodeSerpent(Depcode):
     r"""Class contains information about input, output, geometry, and
     template files for running Serpent2 depletion simulations.
     Also contains neutrons population, active, and inactive cycles.
     Contains methods to read template and output files,
     write new input files for Serpent2.
 
+    Attributes
+    -----------
+    param : dict of str to type
+        Holds Serpent depletion step parameter information
+    sim_info : dict of str to type
```

Comment:
Same comment. Not clear what str and type are

Reply:
I expanded the docstring.